pkg/controllermanager/controller/project/activity/add.go

```diff
@@ -51,16 +52,13 @@ func (r *Reconciler) AddToManager(mgr manager.Manager) error {
 		r.Clock = clock.RealClock{}
 	}
 
-	// It's not possible to overwrite the event handler when using the controller builder. Hence, we have to build up
-	// the controller manually.
-	c, err := controller.New(
-		ControllerName,
-		mgr,
-		controller.Options{
-			Reconciler:              r,
+	c, err := builder.
+		ControllerManagedBy(mgr).
+		Named(ControllerName).
+		WithOptions(controller.Options{
 			MaxConcurrentReconciles: pointer.IntDeref(r.Config.ConcurrentSyncs, 0),
-		},
-	)
+		}).
+		Build(r)
```

Comment:
Can you use `.Watches()` now and drop below lines? And then run `Complete()` instead of `Build`? Or does this still not work?
(same for all other controllers)

Reply:
We pass `c.GetLogger()` in the mapper functions, how do we pass this in the `Watches()`, because the controller won't have been built.

Reply:
OK, let's keep it this way then for now. It's already a first improvement.